core/src/main/java/org/apache/accumulo/core/client/mapred/AbstractInputFormat.java

```diff
@@ -384,7 +387,21 @@ public static InputTableConfig getInputTableConfig(JobConf job, String tableName
   protected abstract static class AbstractRecordReader<K,V> implements RecordReader<K,V> {
     protected long numKeysRead;
     protected Iterator<Map.Entry<Key,Value>> scannerIterator;
-    protected RangeInputSplit split;
+    protected org.apache.accumulo.core.client.mapreduce.impl.AccumuloInputSplit split;
+    protected ScannerBase scannerBase;
+
+    /**
+     * Configures the iterators on a scanner for the given table name.
+     *
+     * @param job
+     *          the Hadoop job configuration
+     * @param scanner
+     *          the scanner for which to configure the iterators
+     * @param tableName
+     *          the table name for which the scanner is configured
+     * @since 1.7.0
+     */
+    protected abstract void setupIterators(JobConf job, ScannerBase scanner, String tableName, AccumuloInputSplit split);
```

Comment:
I suppose `setupIterators` would be considered public API.  However it has parameter of type `AccumuloInputSplit` which is not public.